### PR TITLE
Jakarta EE 10 Update - Servlet API 6.0 & JSP API 3.1

### PR DESCRIPTION
**Pull request by joakime**
This brings the `jetty-jakarta-servlet-api` and `jetty-schemas` up to date.

It also helps with building the `jetty-12.0.x-hackathon` branch.

### CODE REVIEW

**Comment by joakime** [comment]
Both of these artifacts have been deployed already to the snapshot repo.

* `org.eclipse.jetty.toolchain:jetty-jakarta-servlet-api:6.0.0-SNAPSHOT`
* `org.eclipse.jetty.toolchain:jetty-schemas:6.0-SNAPSHOT`

File: jetty-jakarta-servlet-api/pom.xml [hunk 1]
```diff
@@ -314,5 +327,15 @@
         <enabled>true</enabled>
       </snapshots>
     </repository>
+    <repository>
+      <id>jakarta.staging</id>
+      <url>https://jakarta.oss.sonatype.org/content/groups/staging</url>
```
Comment (joakime) [inline jetty-jakarta-servlet-api/pom.xml:332]: FYI, the staged jakarta artifacts can be found at this repo.

Comment (gregw) [reply jetty-jakarta-servlet-api/pom.xml:332]: Can we add TODOs on things that should be removed once the real release is available.

**Review by gregw — CHANGES_REQUESTED** [review]

File: jetty-jakarta-servlet-api/pom.xml (hunk 1 above)
Comment (joakime) [reply jetty-jakarta-servlet-api/pom.xml:334]: @gregw done.

File: jetty-schemas/pom.xml [hunk 2]
```diff
@@ -36,8 +36,8 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <configuration>
-          <source>1.8</source>
-          <target>1.8</target>
+          <source>11</source>
```
Comment (olamy) [inline jetty-schemas/pom.xml:39]: why not simply using `<release>11</release>`?

**Comment by joakime** [comment]
No longer relevant, as the official [`jakarta.servlet:jakarta.servlet-api:6.0`](https://search.maven.org/artifact/jakarta.servlet/jakarta.servlet-api/6.0.0/jar) is now on maven central.

**Pull request closed without merging**